### ISSUE TITLE
Need help overriding styles

### ISSUE THREAD

**Issue by jzissu** [kekeh/mydatepicker#118]
First, let me say that your datepicker has been working wonderfully for me. It's easy to use in my code, very intuitive for the user, and incorporating the events your module emits has been easy to integrate with my application, making the development smooth and painless.

That said, I have one problem which I have not been able to figure out - how to override styles with my own. According to your documentation (and some responses to other issues), I need to make a separate CSS file and load it "after the datepicker has been loaded." Now when I try to put the changes in my index.html file (like your example shows) nothing happens because I haven't loaded the page with the datepicker in it yet. When I navigate to that page, the "override CSS" doesn't get applied, most likely because it was loaded before the datepicker (not after) so it gets the default styling applied.

Where EXACTLY in my Angular2 application should I be loading this file? Specifically, if I have a component which will not load until I navigate to a particular route, how can I be sure to load the "override CSS" after the component is loaded?

**Comment by kekeh** [comment]
Try to put the code below to the end of the **template** which contain the **my-date-picker** tag. Then it is loaded after the mydatepicker is loaded.
```javascript
<style>
    @import "./override.css";
</style>
```
Check the path to the **override.css** file. If the @import is not working you can write style definitions inside the **style** tag.

Let me know if this is working.

If it is not working then you have to fork the mydatepicker component and edit styles from the https://github.com/kekeh/mydatepicker/blob/master/src/my-date-picker/my-date-picker.component.css file. Here are instructions: https://github.com/kekeh/mydatepicker#development-of-this-component

When you edit the styles and execute the **gulp all** command, **npmdist** folder is created. Go to that folder and execute **npm pack**. It creates a local npm package to this folder. For example: mydatepicker-1.0.38.tgz.  

This can be installed easily: **npm install path-to-npmdist/mydatepicker-1.0.38.tgz**

**Comment by jzissu** [comment]
Well, because it's using Angular2 (and the Angular CLI) there is a problem where the attribute is applied to the styling, even when I put the style definition directly in the style tag. However, I did find another way to solve the problem: I used the AfterViewInit lifecycle hook to style the element after it is loaded, since it is considered a "child" of the component where it's embedded. That allowed me to use the following to style the selector to be centered in its enclosing div:

`ngAfterViewInit() {
    let chk: any = document.getElementById('idtag').getElementsByClassName('selector');
    chk[0].style.margin = 'auto';
  }`

**Comment by CliffyMk** [comment]
just add !important to override the inline css

.my-dp{
color:red !important;
}

**Comment by hesparza** [comment]
I tried with !important and it doesn't work for me..

**Comment by CliffyMk** [comment]
Ill send u a code sample 

I have achieved the same 

Also I overrided image for calendar
(I am. Travelling so might take time)

Meanwhile it would be helpful if  you can share your code

Thanks 
Manish Kumar

**Comment by hesparza** [comment]
so !important  didn't work for me, but I managed to solve my problem by using the "**piercing CSS combinators**", specifically I used **/deep/** as apparently is the one that works when you use Angular CLI like in my case. 

Here is an example of the CSS I added to the parent's style file:

> :host /deep/ .mydp {
> /**  My override css properties here **/
> }
> 
> :host /deep/ .mydp, .mydp .headertodaybtn, .mydp .monthinput, .mydp .yearinput {
> /**  My override css properties here **/
> }
> 
> :host /deep/ .mydp * {
> /**  My override css properties here **/
> }

I came with this after reading Chrillewoodz's answer in the next stack overflow's question: [how-to-style-child-components-from-parent-components-css-file](http://stackoverflow.com/questions/36527605/how-to-style-child-components-from-parent-components-css-file)

I found this solution cleaner than using the AfterViewInit life cycle hook. Moreover, it is easier and quicker than having to build and install the mydatepicker project just to accommodate style to a particular case. 

I hope this works for anybody who runs in the same problem.

**Comment by CliffyMk** [comment]
Ill post my code tomorrow .for sure

Manish Kumar

**Comment by chris93007** [comment]
Same issue!

**Comment by hesparza** [comment]
Did you tried using "piercing CSS combinators"??

**Comment by chris93007** [comment]
I tried adding - 
    :host /deep/ .mydp {
        font-size: 14px !important;
        font-weight: 500 !important;
        color: #55595c !important;
    }
 in style tags in the same template file and still no luck.
My project uses scss, could that  be a factor?

**Comment by hesparza** [comment]
I don't think so since the SCSS file is turned into a regular CSS file at the end of the day if I'm not mistaking. 

Did you tried ">>>" instead of "/deep/"?, if you could provide an example of your code that would help a lot

**Comment by chris93007** [comment]
HTML
`                          <my-date-picker [options]="myDatePickerOptions"
                                          (dateChanged)="onDateChanged($event)">
                          </my-date-picker>`

SCSS
`    .mydp {
        font-size: 14px !important;
        font-weight: 500 !important;
        color: #55595c !important;
    }`
   ` .btnpicker{
        width:35px !important;
        background: #d1d4d7 !important;
    }`

I'm basically playing the calendar icon to make it look more like bootstrap's input groups

**Comment by chris93007** [comment]
I tried '>>>' and '/deep/' but scss throws error :/

**Comment by chris93007** [comment]
Finally  had to solve it as such - 
````
@Component({
  selector: 'app-mycomponent',
  templateUrl: './mycomponent.component.html',
  styleUrls: ['./mycomponent.component.scss'],
  styles:[
    `
       :host >>> my-date-picker >>> .mydpicon {
          font-size: 14px !important;
          font-weight: 500 !important;
          color: #55595c !important;
        }
       :host >>> my-date-picker >>> .btnpicker {
          width: 35px;
          background: #d1d4d7;
        }
    `
  ]
})
````

It isn't the neatest solution however, there should be a cleaner way of overwriting styles in an scss or css file.

**Comment by NaLLiFFuNT** [comment]
As a rule, every input fields of application looks alike. So you can put styles for my-date-picker into global css file of your application it works fine for me.

**Comment by jrymer** [comment]
Has anyone managed to change the icons used?

**Comment by dan-pointchain** [comment]
> HTML
> ` <my-date-picker [options]="myDatePickerOptions" (dateChanged)="onDateChanged($event)"> </my-date-picker>`
> 
> SCSS
> ` .mydp { font-size: 14px !important; font-weight: 500 !important; color: #55595c !important; }`
> ` .btnpicker{ width:35px !important; background: #d1d4d7 !important; }`
> 
> I'm basically playing the calendar icon to make it look more like bootstrap's input groups

For me also same issue. Can you help me how to apply mydatepicker style through 
 scss file

**Comment by Prasad-Chowdary** [comment]
> Finally had to solve it as such -
> 
> ```
> @Component({
>   selector: 'app-mycomponent',
>   templateUrl: './mycomponent.component.html',
>   styleUrls: ['./mycomponent.component.scss'],
>   styles:[
>     `
>        :host >>> my-date-picker >>> .mydpicon {
>           font-size: 14px !important;
>           font-weight: 500 !important;
>           color: #55595c !important;
>         }
>        :host >>> my-date-picker >>> .btnpicker {
>           width: 35px;
>           background: #d1d4d7;
>         }
>     `
>   ]
> })
> ```
> 
> It isn't the neatest solution however, there should be a cleaner way of overwriting styles in an scss or css file.


Me also facing same issue . can you help me, how to apply customize styles through scss file

**Comment by kekeh** [comment]
If your Angular version is >= 7.2 then you can use new mydatepicker: https://github.com/kekeh/angular-mydatepicker

Demo page: https://kekeh.github.io/angular-mydatepicker

Then you can easily override the styles of a datepicker. You can test that in the demo page. README contain examples on how to override.

**Comment by Prasad-Chowdary** [comment]
> If your Angular version is >= 7.2 then you can use new mydatepicker: https://github.com/kekeh/angular-mydatepicker
> 
> Demo page: https://kekeh.github.io/angular-mydatepicker
> 
> Then you can easily override the styles of a datepicker. You can test that in the demo page. README contain examples on how to override.

we can change calender icon also ?

**Comment by kekeh** [comment]
That datepicker doesn't contain calendar icon at all. Check from the demo page.

**Comment by Prasad-Chowdary** [comment]
> That datepicker doesn't contain calendar icon at all. Check from the demo page.

Thank you
